tests/projects/backend/test_local.py

```diff
@@ -4,67 +4,75 @@
 from mlflow.projects.backend.local import _get_docker_artifact_storage_cmd_and_envs
 
 
-def test_docker_s3_artifact_cmd_and_envs_from_env():
+def test_docker_s3_artifact_cmd_and_envs_from_env(monkeypatch):
     mock_env = {
         "AWS_SECRET_ACCESS_KEY": "mock_secret",
         "AWS_ACCESS_KEY_ID": "mock_access_key",
         "MLFLOW_S3_ENDPOINT_URL": "mock_endpoint",
         "MLFLOW_S3_IGNORE_TLS": "false",
     }
-    with mock.patch.dict("os.environ", mock_env), mock.patch(
-        "posixpath.exists", return_value=False
-    ):
+    monkeypatch.setenvs(mock_env)
+    with mock.patch("posixpath.exists", return_value=False):
         cmds, envs = _get_docker_artifact_storage_cmd_and_envs("s3://mock_bucket")
         assert cmds == []
         assert envs == mock_env
 
 
-def test_docker_s3_artifact_cmd_and_envs_from_home():
+def test_docker_s3_artifact_cmd_and_envs_from_home(monkeypatch):
     mock_env = {}
-    with mock.patch.dict("os.environ", mock_env), mock.patch(
-        "posixpath.exists", return_value=True
-    ), mock.patch("posixpath.expanduser", return_value="mock_volume"):
+    monkeypatch.setenvs(mock_env)
```

Comment:
monkeypatch and mock work a bit differently. This line does nothing. You will need to explicitly delete the four env vars:

`AWS_SECRET_ACCESS_KEY`, `AWS_ACCESS_KEY_ID`, `MLFLOW_S3_ENDPOINT_URL`, `MLFLOW_S3_IGNORE_TLS`

Reply:
Thank you for the review : )
Ok, understand